### DISCUSSION TITLE
Re using the native search functionality in Quarto, on other parts of the site.

### DISCUSSION THREAD

**Discussion by dweng0** [quarto-dev/quarto-cli#5944] [Q&A]
### Description

I would like to add a search input box to the index page of my Quarto site that utilizes the native search functionality. (think like... google and sites like that, with one search box on the index page) Is it possible to reuse the Search.json in Quarto for this purpose? I couldn't find any documentation on it aside from how to show/hide and display results for the nav bar search.

Any help would be greatly appreciated.

**Comment by mcanouil** [comment]
I have no idea on this one, but @dragonstyle might have one or at least an answer.

**Comment by dragonstyle** [comment]
We currently don't support you manually contributing content to the search index - we automatically generated that for HTML documents produced by Quarto. Just to be sure that I understand - is the idea that you have existing HTML content that you'd like to include in the search index?

**Reply by dweng0** [reply]
It was more to see if It could be leveraged by other documents  like the home page, whereby an input box could be used to "trigger" the search functionality.

**Reply by dragonstyle** [reply]
I see - there currently isn't a way for you to control the positioning of the input (aside from placing it on the navbar or sidebar using `type: textbox`.

**Reply by PeterJacko** [reply]
@dragonstyle It would be really useful to be able to place the search input box on the pages, not only on the navbar and sidebar. We would like to create the landing page that has a search box in the middle of the landing page, like Google. It's a natural thing for webpage visitors. There is a similar request here: https://stackoverflow.com/questions/76485966/in-quarto-is-there-a-way-to-include-the-searchbox-on-the-index-page .

**Reply by mcanouil** [reply]
@PeterJacko Please avoid tagging people when replying in threads.
Since this thread was a question and it has been answered. I would suggest you open a Feature request in discussion explaining what you want and why you want it.

Note that "It's a natural thing for webpage visitors" is your opinion and not a fact.

Finally, you could (in theory) use JavaScript to move the search component on your page. That's not pretty but it should work as a workaround.

**Reply by saumil-sh** [reply]
Hello, I want to createa a wiki / knowledgebase using quarto website (I know it is not ideal). I want to add a search bar in the middle of the index / homepage. I can inspcet and copy the html of `textbox` / `overlay` searchbar and place it where I want. However, this doesn't have the same functionality, i.e. no results appear as I type a search string. Could anyone please help me set this up correctly?

<img width="1522" height="216" alt="image" src="https://github.com/user-attachments/assets/cc42b7a4-18a7-496c-9060-0eb4bda497de" />

<img width="1518" height="217" alt="image" src="https://github.com/user-attachments/assets/907001ac-7fc3-4922-8842-cd6ef15d9c9d" />

EDIT: added images of the intended behavior

**Reply by saumil-sh** [reply]
I was able to move it using javascript successfully. The styling is not perfect, and is work in progress. 
<details>
<summary> Add this at the end of your body. </summary>

```html
<script>
console.log('Search mover script is running');
// Move Quarto's search (#quarto-search) from navbar to the homepage hero.
// We wait until Quarto initializes it (ensuring handlers & autocompletion stay intact).
(function () {
  const MAX_TRIES = 50;   // ~5s total at 100ms intervals
  const INTERVAL  = 100;

  function isSearchReady(el) {
    // Quarto attaches input + button inside. Check for an input to be safe.
    return el && el.querySelector('input[type="search"], input[type="text"]');
  }

  function tryMoveSearch(tries = 0) {
    const search = document.getElementById('quarto-search');
    const slot   = document.getElementById('home-search');

    if (!slot) return; // only run on homepage where slot exists

    if (search && isSearchReady(search)) {
      // Move the actual node so all event listeners remain intact
      slot.appendChild(search);

      // Add a class so we can style it differently on the homepage
      search.classList.add('home-search');

      // Hide the original placeholder area in the navbar (container may vary by theme)
      // We hide any empty .nav-item that used to hold search.
      const nav = document.querySelector('header.navbar, .navbar, header#quarto-header');
      if (nav) {
        // If Quarto wrapped search in a container, hide that container when empty
        const parent = nav.contains(search) ? nav : null;
        if (parent) {
          // nothing; search is already moved. Some themes leave a gap—collapse via CSS below.
        }
      }
      return;
    }

    if (tries < MAX_TRIES) {
      setTimeout(() => tryMoveSearch(tries + 1), INTERVAL);
    } else {
      console.warn('Quarto search did not initialize in time.');
    }
  }

  if (document.readyState === 'loading') {
    document.addEventListener('DOMContentLoaded', tryMoveSearch);
  } else {
    tryMoveSearch();
  }
})();
</script>
```
</details>

<img width="1911" height="537" alt="image" src="https://github.com/user-attachments/assets/2848c290-def8-40d4-8390-3b8599d1d5e2" />
